fix(pkg/cli): ensure correct configuration from kubeconfig file

adamreese opened the pull request:
Bind Helm flags to Kubernetes configuration loader to get a merged
config with kubeconfig.

Fixes: #7539

adamreese commented on pkg/cli/environment.go:
```diff
@@ -107,42 +111,27 @@ func (s *EnvSettings) EnvVars() map[string]string {
-		"HELM_KUBEAPISERVER":     s.KubeAPIServer,
-	}
 
+		// broken, these are populated from helm flags and not kubeconfig.
+		"HELM_KUBECONTEXT":   s.KubeContext,
+		"HELM_KUBETOKEN":     s.KubeToken,
+		"HELM_KUBEAPISERVER": s.KubeAPIServer,
```
Not sure what to do with this

technosophos replied:
Same... I don't understand their purpose in the first place. I guess leaving them alone is the safest bet.

technosophos reviewed (approved):
LGTM. cobra.OnInitialize is GREAT

adamreese committed:
fix(pkg/cli): ensure correct configuration from kubeconfig file

Bind Helm flags to Kubernetes configuration loader to get a merged
config with kubeconfig.

Fixes: #7539

Signed-off-by: Adam Reese <adam@reese.io>

technosophos reviewed (approved):
Re-reviewed with the changes to the unit tests.
